librz/util/subprocess.c

```diff
@@ -963,14 +972,38 @@ RZ_API RzSubprocess *rz_subprocess_start_opt(RzSubprocessOpt *opt) {
 		proc->stderr_fd = proc->stdout_fd;
 	}
 
+no_pipes:
 	// Let's create the environment for the child in the parent, with malloc,
 	// because we can't use functions that lock after fork
 	child_env = create_child_env(opt->envvars, opt->envvals, opt->env_size);
 
-	proc->pid = rz_sys_fork();
+	int master_fd = -1;
+	if (opt->fork_mode == RZ_SUBPROCESS_FORKPTY) {
+		void *term_params = NULL;
+#if HAVE_FORKPTY && HAVE_OPENPTY && HAVE_LOGIN_TTY
+		struct termios t;
+		/* Needed to avoid reading back the writes again from the TTY */
+		tcgetattr(STDIN_FILENO, &t);
+		cfmakeraw(&t);
+		term_params = &t;
+#endif
+		proc->pid = rz_sys_forkpty(&master_fd, NULL, term_params, NULL);
+		proc->master_fd = master_fd;
+
+		proc->stderr_fd = (proc->stderr_fd == -1) ? master_fd : proc->stderr_fd;
+		proc->stdout_fd = (proc->stdout_fd == -1) ? master_fd : proc->stdout_fd;
+		proc->stdin_fd = (proc->stdin_fd == -1) ? master_fd : proc->stdin_fd;
```

Comment:
I guess stderr_fd and the others are always -1 if we are in this branch. Maybe just put a `rz_warn_if_fail(proc->stderr_fd == 1);` etc. just to be sure, but it should never happen